src/gl/directx11/vcRenderShaders.cpp

```diff
@@ -1405,3 +1405,326 @@ const char *const g_udGPURenderGeomGeometryShader = R"shader(
+            if(!doneN) posN.x -= offNP.x * FXAA_QUALITY__P3;
+            if(!doneN) posN.y -= offNP.y * FXAA_QUALITY__P3;
+            doneNP = (!doneN) || (!doneP);
+            if(!doneP) posP.x += offNP.x * FXAA_QUALITY__P3;
+            if(!doneP) posP.y += offNP.y * FXAA_QUALITY__P3;
+            if(doneNP) {
+                if(!doneN) lumaEndN = FxaaLuma(FxaaTexTop(tex, posN.xy));
+                if(!doneP) lumaEndP = FxaaLuma(FxaaTexTop(tex, posP.xy));
+                if(!doneN) lumaEndN = lumaEndN - lumaNN * 0.5;
+                if(!doneP) lumaEndP = lumaEndP - lumaNN * 0.5;
+                doneN = abs(lumaEndN) >= gradientScaled;
+                doneP = abs(lumaEndP) >= gradientScaled;
+                if(!doneN) posN.x -= offNP.x * FXAA_QUALITY__P4;
+                if(!doneN) posN.y -= offNP.y * FXAA_QUALITY__P4;
+                doneNP = (!doneN) || (!doneP);
+                if(!doneP) posP.x += offNP.x * FXAA_QUALITY__P4;
+                if(!doneP) posP.y += offNP.y * FXAA_QUALITY__P4;
+             
+            }
+        }
+    }
+    FxaaFloat dstN = posM.x - posN.x;
+    FxaaFloat dstP = posP.x - posM.x;
+    if(!horzSpan) dstN = posM.y - posN.y;
+    if(!horzSpan) dstP = posP.y - posM.y;
+    FxaaBool goodSpanN = (lumaEndN < 0.0) != lumaMLTZero;
+    FxaaFloat spanLength = (dstP + dstN);
+    FxaaBool goodSpanP = (lumaEndP < 0.0) != lumaMLTZero;
+    FxaaFloat spanLengthRcp = 1.0/spanLength;
+    FxaaBool directionN = dstN < dstP;
+    FxaaFloat dst = min(dstN, dstP);
+    FxaaBool goodSpan = directionN ? goodSpanN : goodSpanP;
+    FxaaFloat subpixG = subpixF * subpixF;
+    FxaaFloat pixelOffset = (dst * (-spanLengthRcp)) + 0.5;
+    FxaaFloat subpixH = subpixG * fxaaQualitySubpix;
+    FxaaFloat pixelOffsetGood = goodSpan ? pixelOffset : 0.0;
+    FxaaFloat pixelOffsetSubpix = max(pixelOffsetGood, subpixH);
+    if(!horzSpan) posM.x += pixelOffsetSubpix * lengthSign;
+    if( horzSpan) posM.y += pixelOffsetSubpix * lengthSign;
+    return FxaaFloat4(FxaaTexTop(tex, posM).xyz, lumaM);
+}
+
+ struct PS_INPUT
+  {
+    float4 pos : SV_POSITION;
+    float2 uv : TEXCOORD0;
+    float2 edgeSampleUV0 : TEXCOORD1;
+    float2 edgeSampleUV1 : TEXCOORD2;
+    float2 edgeSampleUV2 : TEXCOORD3;
+    float2 edgeSampleUV3 : TEXCOORD4;
+    float2 sampleStepSize : TEXCOORD5;
+  };
+
+  struct PS_OUTPUT
+  {
+    float4 Color0 : SV_Target;
+  };
+
+  sampler sampler0;
+  Texture2D texture0;
+
+  sampler sampler1;
+  Texture2D texture1;
+
+  PS_OUTPUT main(PS_INPUT input)
+  {
+    PS_OUTPUT output;
+    float depth = texture1.Sample(sampler1, input.uv).x;
+
+    // only run FXAA on edges (simple edge detection)
+    float depth0 = texture1.Sample(sampler1, input.edgeSampleUV0).x;
+    float depth1 = texture1.Sample(sampler1, input.edgeSampleUV1).x;
+    float depth2 = texture1.Sample(sampler1, input.edgeSampleUV2).x;
+    float depth3 = texture1.Sample(sampler1, input.edgeSampleUV3).x;
```

Comment:
Are the parameters for this sampler set-up to clamp?

Reply:
Yep!